Android_Development_with_Kotlin/09. Dynamic User Interface/09.13 Date Picker.md

```diff
@@ -0,0 +1,174 @@
+| android:maxDate                     |  The maximal date shown by this calendar view in mm/dd/yyyy format.         |
+| android:minDate                     |  The minimal date shown by this calendar view in mm/dd/yyyy format.   |
+| android:spinnersShown               |  Whether the spinners are shown.                            |
+| android:startYear                 	 |  The first year (inclusive), for example "1940".      |
+| android:yearListItemTextAppearance  |  The list year's text appearance in the list.               |
+| android:yearListSelectorColor       |  The list year's selected circle color in the list.    |
+
+<hr>
+
+# Tutorial
+
+## To use Calendar DatePicker in activity_main.xml
+
+In this file, we will add the DatePicker and Button widget and set their attributes so that it can be accessed in the kotlin file. 
+ 
+ ```kotlin
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+	android:orientation="vertical"
+	android:layout_width="match_parent"
+	android:layout_height="match_parent"
+	android:id="@+id/linear_layout"
+	android:gravity = "center">
+
+	<DatePicker
+		android:id="@+id/date_Picker"
+		android:layout_width="match_parent"
+		android:layout_height="match_parent"/>
+</LinearLayout>
+ ```
+
+ ## To use Spinner DatePicker in activity_main.xml
+
+ In this file, we will add the DatePicker and Button widget and set their attributes so that it can be accessed in the kotlin file. 
+
+  ```kotlin
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+	android:orientation="vertical"
+	android:layout_width="match_parent"
+	android:layout_height="match_parent"
+	android:id="@+id/linear_layout"
+	android:gravity = "center">
+
+	<DatePicker
+		android:id="@+id/date_Picker"
+		android:layout_width="match_parent"
+		android:layout_height="match_parent"
+		android:datePickerMode = "spinner"
+		android:calendarViewShown="false"/>
+</LinearLayout>
+   ```
+
+## Access the DatePicker in MainActivity.kt file
+
+First of all, we declare a variable datePicker to access the DatePicker widget from the XML layout. 
+
+```kotlin
+val datePicker = findViewById<DatePicker>(R.id.date_Picker)
+```
+
+then, we declare another variable today to get the current get like this. 
+
+```kotlin
+val today = Calendar.getInstance()
+    datePicker.init(today.get(Calendar.YEAR), today.get(Calendar.MONTH),
+    today.get(Calendar.DAY_OF_MONTH)
+```
+
+To display the selected date from the calendar we will use 
+
+```kotlin
+{ view, year, month, day ->
+            val month = month + 1
+            val msg = "You Selected: $day/$month/$year"
+            Toast.makeText(this@MainActivity, msg, Toast.LENGTH_SHORT).show()
+ }
+ ```
+
+ We are familiar with further activities in previous articles like accessing button and set OnClickListener etc. 
+
+ ```kotlin
+package com.geeksforgeeks.myfirstkotlinapp

```

Comment:
You must not use any other website name in the code, please rename the package name

Reply:
okay sir , I've changed , you may check now.